contracts/FIFOMarket.sol

```diff
@@ -242,6 +283,51 @@ contract FIFOMarket is
     _removal.burnBatch(address(this), batchedIds, batchedAmounts);
   }
 
+  // TODO batch version of this?
+  function reserveRemoval(uint256 removalId) external returns (bool) {
+    address supplierAddress = removalId.supplierAddress();
+    EnumerableSetUpgradeable.UintSet storage supplierSet = _activeSupply[
+      supplierAddress
+    ];
+    require(
+      supplierSet.contains(removalId),
+      "Market: removal not in active supply"
+    );
+    supplierSet.remove(removalId);
+    totalNumberActiveRemovals -= 1;
+    uint256 removalBalance = _removal.balanceOf(address(this), removalId);
+    totalActiveSupply -= removalBalance;
+    totalReservedSupply += removalBalance;
+    // If this is the last removal for the supplier, remove them from active suppliers
+    if (supplierSet.length() == 0) {
+      _removeActiveSupplier(supplierAddress);
+    }
+    // todo any checks on whether this id was already in there?
+    return _reservedSupply.add(removalId); // returns true if the value was added to the set, that is, if it was not already present
+  }
+
+  // TODO batch version of this?
+  function unreserveRemoval(uint256 removalId) external returns (bool) {
+    address supplierAddress = removalId.supplierAddress();
+    EnumerableSetUpgradeable.UintSet storage supplierSet = _activeSupply[
+      supplierAddress
+    ];
+    require(
+      _reservedSupply.contains(removalId),
+      "Market: removal not in reserved supply"
+    );
+    _reservedSupply.remove(removalId);
```

Comment:
Since [`.remove`](https://docs.openzeppelin.com/contracts/4.x/api/utils#EnumerableSet-remove-struct-EnumerableSet-UintSet-uint256-) returns true if the value was removed (and therefore present) the following should be sufficient:
```suggestion
    require(
      _reservedSupply.remove(removalId) == true,
      "Market: removal not in reserved supply"
    );
    ```